### PR TITLE
integration test for validator to delegation

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Integration tests for new "validator to delegation manager" functionalities

### CODE REVIEW

File: integrationTests/frontend/staking/staking_test.go [hunk 1]
```diff
@@ -139,3 +425,46 @@ func checkStakeOnNode(t *testing.T, n *integrationTests.TestProcessorNode, blsKe
 	require.Equal(t, 1, len(vmOutput.ReturnData))
 	assert.Equal(t, []byte("staked"), vmOutput.ReturnData[0])
 }
+
+const delegationManagementKey = "delegationManagement"
```
Comment (iulianpascalau) [inline integrationTests/frontend/staking/staking_test.go:429]: these should be placed in the first part of the file

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
Managed to reduce the duplicated code inside the integration tests. Will open a PR against this one

File: integrationTests/frontend/staking/staking_test.go [hunk 2]
```diff
@@ -118,6 +121,289 @@ func TestSignatureOnStaking(t *testing.T) {
...
+	saveDelegationContractsList(nodes)
+
+	maxDelegationCap := []byte{0}
+	serviceFee := []byte{10}
+	txData = txDataBuilder.NewBuilder().Clear().
+		Func("makeNewContractFromValidatorData").
+		Bytes(maxDelegationCap).
+		Bytes(serviceFee).
+		ToString()
+	//txData = "makeNewContractFromValidatorData" + "@" + hex.EncodeToString() + "@" + pubKey + "@" + frontendHexSignature
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/frontend/staking/staking_test.go]: done

File: integrationTests/frontend/staking/staking_test.go [hunk 3]
```diff
@@ -118,6 +121,289 @@ func TestSignatureOnStaking(t *testing.T) {
...
+	saveDelegationContractsList(nodes)
+
+	maxDelegationCap := []byte{0}
+	serviceFee := []byte{0}
+	txData = txDataBuilder.NewBuilder().Clear().
+		Func("createNewDelegationContract").
+		Bytes(maxDelegationCap).
+		Bytes(serviceFee).
+		ToString()
+	//txData = "makeNewContractFromValidatorData" + "@" + hex.EncodeToString() + "@" + pubKey + "@" + frontendHexSignature
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/frontend/staking/staking_test.go]: done

File: integrationTests/frontend/staking/staking_test.go (hunk 1 above)
Comment (iulianpascalau) [reply integrationTests/frontend/staking/staking_test.go]: done

**Review by iulianpascalau — APPROVED** [review]
good cleanup :+1:

File: integrationTests/frontend/staking/staking_test.go [hunk 4]
```diff
@@ -118,6 +121,289 @@ func TestSignatureOnStaking(t *testing.T) {
 	testStakingWasDone(t, nodes, frontendBLSPubkey)
 }
 
+func TestValidatorToDelegationManagerWithNewContract(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply integrationTests/frontend/staking/staking_test.go]: done

**Review by sasurobert — APPROVED** [review]

**Pull request merged**